src/app/vault/ciphers.component.ts

```diff
@@ -80,6 +86,96 @@ export class CiphersComponent extends BaseCiphersComponent implements OnDestroy
         this.didScroll = this.pagedCiphers.length > this.pageSize;
     }
 
+    async loadOrganizationNames(): Promise<Map<string, string>> {
+        const organizations = await this.userService.getAllOrganizations();
+        const organizationNames = new Map<string, string>();
+
+        if (organizations.length > 0) {
+            organizations.forEach(organization => {
+                organizationNames.set(organization.id, organization.name);
+            });
+        }
+
+        return organizationNames;
+    }
+
+    getOrganizationById(c: CipherView): string {
+        if (c.organizationId != null) {
+            return this.organizationNames.get(c.organizationId);
+        }
+        return this.i18nService.t('myVault');
+    }
+
+    sortCiphers(sortBy: string): void {
+        if (this.sortByOptions.includes(sortBy)) {
+            if (sortBy === this.sortBy) {
+                this.sortedDescending = !this.sortedDescending;
+            } else {
+                this.sortBy = sortBy;
+                this.sortedDescending = true;
+            }
+
+            const self = this;
+
+            if (sortBy === 'lastEdited') {
+                this.ciphers.sort((a: CipherView, b: CipherView) => {
+                    if (self.sortedDescending) {
+                       return new Date(b.revisionDate).valueOf() - new Date(a.revisionDate).valueOf();
+                    } else {
+                        return new Date(a.revisionDate).valueOf() - new Date(b.revisionDate).valueOf();
+                    }
+                });
+            } else if (sortBy === 'dateCreated') {
+                // TO-DO
+            } else if (sortBy === 'owner') {
+                const sortingArray = this.organizationNames;
+
+                this.ciphers.sort((a: CipherView, b: CipherView) => {
+                    let sortingValue1;
+                    let sortingValue2;
+
+                    if (a.organizationId == null) {
+                        sortingValue1 = 'Personal';
+                    } else {
+                        sortingValue1 = sortingArray.get(a.organizationId);
+                    }
+
+                    if (b.organizationId == null) {
+                        sortingValue2 = 'Personal';
+                    } else {
+                        sortingValue2 = sortingArray.get(b.organizationId);
+                    }
+
+                    if (self.sortedDescending) {
+                        return sortingValue1.localeCompare(sortingValue2);
```

Comment:
We have a helper function in `Utils.getSortFunction` which I think you should use here. You can look for examples in the code as to how it's used.

Reply:
I'm actually not sure how to correctly implement this so help here would be appreciated.

Reply:
I found some cipher sorting functions in `cipherService` that I didn't know about 😬  I've included more information and an example in my new review (incoming shortly).